miden-lib/tests/test_note_scripts.rs

```diff
@@ -0,0 +1,646 @@
+    // --------------------------------------------------------------------------------------------
+    let data_store_failure = MockDataStore::with_existing(
+        Some(sender_account.clone()),
+        Some(vec![note_too_late.clone()]),
+    );
+    let mut executor_failure = TransactionExecutor::new(data_store_failure.clone());
+
+    executor_failure.load_account(sender_account_id).unwrap();
+
+    let block_ref = data_store_failure.block_header.block_num().as_int() as u32;
+    let note_origins_failure = data_store_failure
+        .notes
+        .iter()
+        .map(|note| note.proof().as_ref().unwrap().origin().clone())
+        .collect::<Vec<_>>();
+
+    // execute the transaction and get the witness
+    let transaction_result_5 = executor_failure
+        .execute_transaction(
+            sender_account_id,
+            block_ref,
+            &note_origins_failure,
+            Some(tx_script.clone()),
+        )
+        .unwrap();
+
+    // Assert that the sender_account received the funds and the nonce increased by 1
+    // nonce delta
+    assert!(transaction_result.account_delta().nonce == Some(Felt::new(2)));
+
+    // vault delta (Note: vault was empty before)
+    let sender_account_after: Account = Account::new(
+        sender_account_id,
+        AccountVault::new(&vec![fungible_asset_1]).unwrap(),
+        sender_account_storage,
+        sender_account_code,
+        Felt::new(2),
+    );
+    assert!(transaction_result_5.final_account_hash() == sender_account_after.hash());
+
+    // CONSTRUCT AND EXECUTE TX (Case "too late" - Malicious Target Account Failure)
+    // --------------------------------------------------------------------------------------------
+    let data_store_failure = MockDataStore::with_existing(
+        Some(malicious_account.clone()),
+        Some(vec![note_too_late.clone()]),
+    );
+    let mut executor_failure = TransactionExecutor::new(data_store_failure.clone());
+
+    executor_failure.load_account(malicious_account_id).unwrap();
+
+    let block_ref = data_store_failure.block_header.block_num().as_int() as u32;
+    let note_origins_failure = data_store_failure
+        .notes
+        .iter()
+        .map(|note| note.proof().as_ref().unwrap().origin().clone())
+        .collect::<Vec<_>>();
+
+    // execute the transaction and get the witness
+    let transaction_result_6 = executor_failure.execute_transaction(
+        malicious_account_id,
+        block_ref,
+        &note_origins_failure,
+        Some(tx_script.clone()),
+    );
+
+    // check that we got the expected result - TransactionExecutorError and not TransactionResult
+    match transaction_result_6 {
+        Ok(_) => {
+            panic!("Sixth transaction should not work (malicious account can never consume), we expect an error");
+        } // expected result, we do nothing
+        Err(_) => {} // expected result, we do nothing
+    }
+}
```

Comment:
use can use something like `assert!(transaction_result_6.is_err())`

Reply:
Jup, do you think this is way better? Happy to change it.

I had the feeling that as it is right now, it has better readability.

Reply:
I personally prefer the more concise version with a comment, something like

```rust
// Sixth transaction should not work (malicious account can never consume), we expect an error
assert!(transaction_result_6.is_err())
```

I personally think this is more readable as it more concisely expresses the same information.

Reply:
I changed it.